tests/tests_tangd_custom_port.yml

```
@@ -0,0 +1,48 @@
+---
+- name: Test tangd_custom_port
+  hosts: all
+  vars:
+    nbde_server_port: 7500
+    nbde_server_firewall_zone: public
+  tasks:
+    - name: install with custom port and firewall zone
+      import_role:
+        name: linux-system-roles.nbde_server
+
+    - name: check if port is open
+      shell:
+        cmd: |-
+          set -euo pipefail
+          ss -tulpn | grep {{ nbde_server_port }} | awk -F' ' '{print $5}'
+      register: __open_ports_output
+      failed_when: __open_ports_output.stdout != "*:{{ nbde_server_port }}"
+
+    - name: check if port TCP is open
+      shell:
+        cmd: |-
+          set -euo pipefail
+          ss -tulpn | grep {{ nbde_server_port }} | awk -F' ' '{print $1}'
+      register: __open_ports_output
+      failed_when: __open_ports_output.stdout != "tcp"
+
+    - name: check if port is opened in firewall
+      shell:
```

```suggestion
      command: firewall-cmd --zone {{ nbde_server_firewall_zone }} --query-port {{ nbde_server_port }}/tcp
      changed_when: false
```
Then you don't need `shell`, `awk`, `grep`, `register`, `failed_when`, etc.

---

yeah ... really not sure why I didn't went for the firewall-cmd command ... glad you found that out. Way more simpler than my solution.